gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/jsr223/JavaTranslator.java

```diff
@@ -273,4 +274,20 @@ private Method getStartMethodFromAnonymousTraversal() {
 
         return null;
     }
+
+    private static final class ReflectedMethod {
+        private final Method method;
+        private final Parameter[] parameters;
+        private final boolean hasVarArgs;
+
+        public ReflectedMethod(final Method m) {
+            this.method = m;
+
+            // the reflection getParameters() method calls clone() every time to get the Parameter array. caching it
+            // saves a lot of extra processing
+            this.parameters = m.getParameters();
+
+            this.hasVarArgs = parameters.length > 0 && parameters[method.getParameters().length - 1].isVarArgs();
```

Comment:
Replace `method.getParameters()` with `parameters`?

Reply:
thanks. i should not have missed that. i'll change it and re-run the benchmark.